ts/packages/anchor/src/program/namespace/types.ts

```diff
@@ -195,10 +195,10 @@ declare type DecodeEnumFields<
  */
 declare type DecodeEnum<K extends IdlTypeDefTyEnum, Defined> = {
   // X = IdlEnumVariant
-  [X in K["variants"][number] as Uncapitalize<X["name"]>]+?: DecodeEnumFields<
+  [X in K["variants"][number] as Uncapitalize<X["name"]>]+?: X["fields"] extends unknown[] ? DecodeEnumFields<
```

Comment:
🍎 if fields exist goes into old branch if not just creating an object from scratch